tests/unit/test_coloring.py

```diff
@@ -0,0 +1,200 @@
+    v2 = None
+
+    @classmethod
+    def create_coloring_helper(self, vertices: List[int], map: dict):
+        coloring = Coloring()
+        for color in map:
+            for value in map[color]:
+                vertex = [v for v in vertices if v.label == value][0]
+                coloring.set(vertex, color)
+        return coloring
+
+    @classmethod
+    def create_graph_helper(self, edges: List[List[int]]):
+        g = Graph(False)
+        vertices = {}
+        for edge in edges:
+            head, tail = edge
+            if head not in vertices:
+                vertices[head] = Vertex(g, label=head)
+                g.add_vertex(vertices[head])
+            if tail not in vertices:
+                vertices[tail] = Vertex(g, label=tail)
+                g.add_vertex(vertices[tail])
+            g.add_edge(Edge(vertices[head], vertices[tail]))
+        return g
+
+    def setUp(self):
+        self.coloring = Coloring()
+        self.graph = Graph(False, 5)
+        self.v1, self.v2 = Vertex(self.graph), Vertex(self.graph)
+
+    def test_init(self):
+        # Has no colors or vertices
+        self.assertEqual(0, len(self.coloring))
+        self.assertEqual(0, len(self.coloring.vertices))
+
+    def test_set(self):
+        # Adds a new color if not existing
+        self.coloring.set(self.v1, 4)
+        self.assertEqual(1, len(self.coloring))
+        self.assertEqual(4, list(self.coloring.colors)[0])
+
+        # Adds no new color if already existing
+        self.coloring.set(self.v2, 4)
+        self.assertEqual(1, len(self.coloring))
+
+        # Cannot set a vertex that is already in the coloring (have to recolor it)
+        with self.assertRaises(KeyError) as e:
+            self.coloring.set(self.v1, 0)
+        self.assertEqual('\'Vertex {} already in coloring, color: 4\''.format(str(self.v1)), str(e.exception))
+
+    def test_get(self):
+        self.coloring.set(self.v1, 4)
+        self.coloring.set(self.v2, 4)
+        vertices = self.coloring.get(4)
+        self.assertEqual(2, len(vertices))
+        self.assertEqual(True, self.v1 in vertices)
+        self.assertEqual(True, self.v2 in vertices)
+
+    def test_add(self):
+        # Adds new vertices
+        self.coloring.add([self.v1,self.v2], 4)
+        vertices = self.coloring.get(4)
+        self.assertEqual(2, len(vertices))
+        self.assertEqual(True, self.v1 in vertices)
+        self.assertEqual(True, self.v2 in vertices)
+
+        # Recolors old vertices
+        v3, v4 = Vertex(self.graph), Vertex(self.graph)
+        self.coloring.set(v3, 4)
+        self.coloring.add([self.v1, self.v2, v4], 0)
+        vertices = self.coloring.get(0)
+        self.assertEqual(3, len(vertices))
+        self.assertEqual(True, self.v1 in vertices)
+        self.assertEqual(True, self.v2 in vertices)
+        self.assertEqual(True, v4 in vertices)
+        self.assertEqual(False, v3 in vertices)
+        self.assertEqual(1, len(self.coloring.get(4)))
```

Comment:
Wil je ook nog testen of het goed gaat als er een vertex in de lijst zit die al in je coloring zit?